release notes/v0.40.0.md

```diff
@@ -0,0 +1,285 @@
+k6 v0.40.0 is here!  This release includes:
+- Breaking changes to some undocumented and unintentional edge behaviors
+- New experimental modules and first-class support for JavaScript classes
+- Bugs and refactorings to pave the way for future features.
+
+Finally,  the [Roadmap](#roadmap-and-future-plans) goes over the plans for the next cycles.
+
+## Breaking changes
+
+- [#2632](https://github.com/grafana/k6/pull/2632) During the refactoring to set tags to `metric.add` in the order they are provided, we discovered that you could provide tags as a key-value pair map *multiple* times in the same call. This was never the intended use and was never documented. As it was undocumented, and as such functionality makes no sense alongside every other API k6 has, we decided to remove this ability.
+- [#2582](https://github.com/grafana/k6/pull/2582) Previously, if `RegisterCallback` result was called twice, the second call would silently break the event loop. This has never been expected behavior, and calling it twice is always a bug in the code using it. Now, calling the `RegisterCallback` result twice leads to a panic.
```

Comment:
```suggestion
```
Should we even mention this in this section, considering it's a Golang change? I think it will mostly confuse users, not help them. I suggest moving it to the `Maintenance and internal improvements` section.

Reply:
I would prefer to preface this with `For extensions`
```suggestion
- [#2582](https://github.com/grafana/k6/pull/2582) [For extensions] Previously, if `RegisterCallback` result was called twice, the second call would silently break the event loop. This has never been expected behavior, and calling it twice is always a bug in the code using it. Now, calling the `RegisterCallback` result twice leads to a panic.
```

Maybe even `For extensions using the event loop` :thinking:

Reply:
>Maybe even For extensions using the event loop

I prefer this version, it makes very clear who can be affected by this

Reply:
I did update the preview with the longer version